File: Frost.Net/Clock.cs
using Frost.Interoperability;
using System.Runtime.CompilerServices;

namespace Frost;

public static class Clock
{
	/// <summary>
	/// Returns the frequency of the clock (ticks per second)
	/// </summary>
	public static ulong Frequency
	{
		[MethodImpl(MethodImplOptions.AggressiveInlining)]
		get => FrostApi.Clock.GetFrequency();
	}

	/// <summary>
	/// Returns the period of the clock (seconds per clock tick)
	/// </summary>
	public static double Period
	{
		[MethodImpl(MethodImplOptions.AggressiveInlining)]
		get => FrostApi.Clock.GetPeriod();
	}

	/// <summary>
	/// Returns current timestamp (ticks)
	/// </summary>
	public static ulong Timestamp
	{
		[MethodImpl(MethodImplOptions.AggressiveInlining)]
		get => FrostApi.Clock.GetTimestamp();
	}
}

File: Frost/_src/clock.cpp
#include "Frost.Api/clock.api.hpp"
#include "../clock.hpp"

namespace frost
{
	u64 clock::get_frequency()
	{
		return api::clock::get_frequency();
	}
	f64 clock::get_period()
	{
		return api::clock::get_period();
	}
	u64 clock::get_timestamp()
	{
		return api::clock::get_timestamp();
	}
}
